test/unit/testExecutorsDealing.js

```diff
@@ -0,0 +1,70 @@
+'use strict';
+
+const initSteps = require('app/core/initSteps');
+const {expect, assert} = require('chai');
+
+describe('Executors-Applying', function () {
+    let ctx;
+    const ExecsDealing = initSteps([__dirname + '/../../app/steps/action/', __dirname + '/../../app/steps/ui']).ExecutorsDealingWithEstate;
+    describe('pruneFormData', () => {
+
+        it('test that isApplying flag is deleted when executor is not applying', () => {
+            ctx = {
+                fullName: 'Ed Brown',
+                isApplying: false
+            };
+            ExecsDealing.pruneFormData(ctx);
+            assert.isUndefined(ctx.isApplying);
+            expect(ctx).to.deep.equal({fullName: 'Ed Brown'});
+        });
+
+        it('test that notApplying data is pruned when executor is applying', () => {
+            ctx = {
+                fullName: 'Ed Brown',
+                isApplying: true,
+                isDead: 'not sure',
+                diedBefore: 'not sure',
+                notApplyingReason: 'not sure',
+                notApplyingKey: 'not sure'
+            };
+            ExecsDealing.pruneFormData(ctx);
+            expect(ctx).to.deep.equal({
+                fullName: 'Ed Brown',
+                isApplying: true
+            });
+        });
+    });
+
+    describe('handlePost', () => {
+        beforeEach(() => {
+            ctx = {
+                list: [
+                    {
+                        'lastName': 'the',
+                        'firstName': 'applicant',
+                        'isApplying': 'Yes',
+                        'isApplicant': true
+                    }, {
+                        fullName: 'Ed Brown',
+                        address: '20 Green Street, London, L12 9LN'
+                    }, {
+                        fullName: 'Dave Miller',
+                        address: '102 Petty Street, London, L12 9LN'
+                    }
+                ],
+                executorsApplying: ['Dave Miller'],
+                executorsNumber: 3
+            };
+        });
+
+        it('test executors (with checkbox unchecked) isApplying flag is deleted', () => {
+            ExecsDealing.handlePost(ctx);
```

Comment:
`[ctx, errors]` should be returned by this method and those values should be tested in the assertions.

`ctx` in `beforeEach()` should be renamed to something like `ctxTest` to prevent a conflict with `ctx` above.